File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        left = tcost = maxlen = 0
        
        for right in range(len(s)):
            tcost += abs(ord(s[right]) - ord(t[right]))
            
            while tcost > maxCost:
                tcost -= abs(ord(s[left]) - ord(t[left]))
                left += 1
            
            maxlen = max(maxlen, right - left + 1)
        
        return maxlen

File: 1208-get-equal-substrings-within-budget/1208-get-equal-substrings-within-budget.rb
# @param {String} s
# @param {String} t
# @param {Integer} max_cost
# @return {Integer}
def equal_substring(s, t, max_cost)
    left = tcost = maxlen = 0
    s.chars.each_index do |right|
        tcost+= (s[right].ord - t[right].ord).abs

        while tcost > max_cost
            tcost -= (s[left].ord - t[left].ord).abs
            left+=1

            maxlen = [maxlen, right - left+1].max

        return maxlen
        
        end
    end
end

